2d_PSF_code/Zernike_Module.py

```diff
@@ -646,6 +654,106 @@ def _addRay(self, pupil, p0, angle, thickness, angleunit=None):
                           & ((self.u - p0[0]) * np.cos(angleRad)
                              + (self.v - p0[1]) * np.sin(angleRad) >= 0)] = True
 
+    def pfiIlum_2d(self, pupil, fiber_id, p0=(0, 0)):
+        """Apply PFI correction to the illumination
+
+        Parameters
+        ----------
+        fiber_id: `int`
+            Fiber id number
+
+        Notes
+        ----------
+        # define distance from the center of the pupil
+        # something like r_dist = sqrt((self.u - p0[0])**2 + (self.v - p0[1])**2), where p0 is the center
+        # first attempt, keep p0 at 0,0
+
+        #   Function to get 1d radial profile for a given fiber (pfiIlum_1d)
+        #    1. take /home/ncaplar/Tickets/PIPE2D-955/subarusb.csv, move to pandas dataframe, find an
+        #    appropriate storage for it (dev_2ddrp); Neven will check if there is a file without FRD or
+        #   if this one is without FRD
+        #   2. given fiber_id, find position on the focal plane - there is a function for that,
+        #  Hassan can help
+        #   3. given radius, find angle (we just did this - it might in pfs_utils already -> check;
+        #    if not, file a ticket)
+        #   4. given angle on the focal plane get theoretical radial profile
+        #      4.b) interpolate radial profile, given input angle
+
+        # apply a function that gives 1d radial profile (pfiIlum_1d), on an array which is a function of
+        # distance from a center of the pupil (r_dist)
+        # pupil.illuminated = pupil.illuminated * pfiIlum_1d(r_dist)
```

Comment:
DocString for this particular function needs to be cleaned up from my working instructions into a proper docstring.